LibraryApp/src/app/components/CustomButton/index.js

```diff
@@ -1,23 +1,62 @@
 import React, { Component } from 'react';
-import { Text, TouchableOpacity, ViewPropTypes, Image } from 'react-native';
+import { Text, TouchableOpacity, ViewPropTypes, Image, Animated } from 'react-native';
 import PropTypes from 'prop-types';
+import { pictonBlue } from '@constants/colors';
+import { DEFAULT_BUTTON_WIDTH } from '@constants/dimensions';
 
 import styles from './styles';
 
+const AnimatedButton = Animated.createAnimatedComponent(TouchableOpacity);
+const ANIM_DURATION = 2000;
+
 class CustomButton extends Component {
+  state = {
+    pressAnim: new Animated.Value(0)
+  };
+
+  onPress = () => {
+    const { onPress } = this.props;
+    Animated.timing(this.state.pressAnim, {
+      toValue: 1,
+      duration: ANIM_DURATION
+    }).start();
+    if (onPress) onPress();
+  };
+
   render() {
-    const { title, icon, whiteTheme, buttonStyles, textStyles, iconStyles, onPress } = this.props;
+    const { title, icon, whiteTheme, buttonStyles, textStyles, iconStyles, onPressStyles } = this.props;
     const titleToUpper = title && title.toUpperCase();
+    const { pressAnim } = this.state;
     return (
-      <TouchableOpacity
-        style={[styles.button, whiteTheme && styles.whiteThemeButton, buttonStyles]}
-        onPress={onPress}
+      <AnimatedButton
+        style={[
+          styles.button,
+          whiteTheme && styles.whiteThemeButton,
+          buttonStyles,
+          onPressStyles?.color && {
+            backgroundColor: pressAnim.interpolate({
+              inputRange: [0, 1],
+              outputRange: [pictonBlue, onPressStyles.color]
+            }),
+            borderColor: pressAnim.interpolate({
+              inputRange: [0, 1],
+              outputRange: [pictonBlue, onPressStyles.color]
+            })
+          },
+          onPressStyles?.width && {
+            width: pressAnim.interpolate({
+              inputRange: [0, 1],
```

Comment:
I would create a constant `inputRange: [0, 1]` and here you will just have to do `inputRange,`